presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/BenchmarkSTArea.java

```diff
@@ -0,0 +1,120 @@
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.plugin.geospatial.GeoFunctions.stGeometryFromText;
+import static com.facebook.presto.plugin.geospatial.GeoFunctions.toSphericalGeography;
+import static com.facebook.presto.plugin.geospatial.GeometryBenchmarkUtils.createPolygon;
+import static com.facebook.presto.plugin.geospatial.GeometryBenchmarkUtils.loadPolygon;
+import static io.airlift.slice.Slices.utf8Slice;
+import static org.testng.Assert.assertEquals;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkSTArea
+{
+    @Benchmark
+    public Object stSphericalArea(BenchmarkData data)
+    {
+        return GeoFunctions.stSphericalArea(data.geography);
+    }
+
+    @Benchmark
+    public Object stSphericalArea500k(BenchmarkData data)
+    {
+        return GeoFunctions.stSphericalArea(data.geography500k);
+    }
+
+    @Benchmark
+    public Object stArea(BenchmarkData data)
+    {
+        return GeoFunctions.stArea(data.geometry);
+    }
+
+    @Benchmark
+    public Object stArea500k(BenchmarkData data)
+    {
+        return GeoFunctions.stArea(data.geometry500k);
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private Slice geometry;
+        private Slice geometry500k;
+        private Slice geography;
+        private Slice geography500k;
+
+        @Setup
+        public void setup()
+                throws IOException
+        {
+            geometry = stGeometryFromText(utf8Slice(loadPolygon("large_polygon.txt")));
+            geometry500k = stGeometryFromText(utf8Slice(createPolygon(9.0, 500000)));
+            geography = toSphericalGeography(geometry);
+            geography500k = toSphericalGeography(geometry500k);
+        }
+    }
+
+    public static void main(String[] args)
+            throws IOException, RunnerException
+    {
+        // assure the benchmarks are valid before running
+        verify();
+
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(".*" + BenchmarkSTArea.class.getSimpleName() + ".*")
+                .build();
+        new Runner(options).run();
+    }
+
+    @Test
+    public static void verify() throws IOException
+    {
+        BenchmarkData data = new BenchmarkData();
+        data.setup();
+        BenchmarkSTArea benchmark = new BenchmarkSTArea();
+
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea(data) / 3.659E8), 1000);
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea500k(data) / 2151E11), 1000);
+        assertEquals(benchmark.stArea(data), 0.05033099592771004);
+        assertEquals(benchmark.stArea500k(data), 0.0);
```

Comment:
I was surprised to see zero here. Then, I took a closer look at the 500K polygon and realized that it's boundary is a great-circle crossing the poles and having all points at a constant longitude (is that right?). It projects into a single line (walked twice) and hence has zero area. I think this is a good polygon to test edge-cases in the area computation, but I wouldn't use it for benchmarking. A polygon with lots of vertexes, but no "gotchas" would be a better fit. Also, `createPolygon` method name sounds fairly generic, but the implementation creates a very particular polygon. Let's move this method out of utility class into this benchmark.

Reply:
the polygon is meant to go around the earth, and include the north pole. In 2d geometry, it ends up not having an area, but does in spherical geometry.